cmd/bitsongd/cmd/config.go

```diff
@@ -0,0 +1,235 @@
+		Short: "Create or query an application CLI configuration file",
+		RunE:  runConfigCmd,
+		Args:  cobra.RangeArgs(0, 2),
+	}
+	return cmd
+}
+
+func runConfigCmd(cmd *cobra.Command, args []string) error {
+	clientCtx := client.GetClientContextFromCmd(cmd)
+	configPath := filepath.Join(clientCtx.HomeDir, "config")
+
+	conf, err := getClientConfig(configPath, clientCtx.Viper)
+	if err != nil {
+		return fmt.Errorf("couldn't get client config: %v", err)
+	}
+
+	tcc := BitsongCustomClient{
+		*conf,
+		os.Getenv("BITSONGD_GAS"),
+		os.Getenv("BITSONGD_GAS_PRICES"),
+		os.Getenv("BITSONGD_GAS_ADJUSTMENT"),
+
+		os.Getenv("BITSONGD_FEES"),
+		os.Getenv("BITSONGD_FEE_GRANTER"),
+		os.Getenv("BITSONGD_FEE_PAYER"),
+
+		os.Getenv("BITSONGD_NOTE"),
+	}
+
+	switch len(args) {
+	case 0:
+		s, err := json.MarshalIndent(tcc, "", "\t")
+		if err != nil {
+			return err
+		}
+
+		cmd.Println(string(s))
+
+	case 1:
+		// it's a get
+		key := args[0]
+
+		switch key {
+		case flags.FlagChainID:
+			cmd.Println(conf.ChainID)
+		case flags.FlagKeyringBackend:
+			cmd.Println(conf.KeyringBackend)
+		case tmcli.OutputFlag:
+			cmd.Println(conf.Output)
+		case flags.FlagNode:
+			cmd.Println(conf.Node)
+		case flags.FlagBroadcastMode:
+			cmd.Println(conf.BroadcastMode)
+
+		// Custom flags
+		case flags.FlagGas:
+			cmd.Println(tcc.Gas)
+		case flags.FlagGasPrices:
+			cmd.Println(tcc.GasPrices)
+		case flags.FlagGasAdjustment:
+			cmd.Println(tcc.GasAdjustment)
+		case flags.FlagFees:
+			cmd.Println(tcc.Fees)
+		case flags.FlagFeeGranter:
+			cmd.Println(tcc.FeeGranter)
+		case flags.FlagFeePayer:
+			cmd.Println(tcc.FeePayer)
+		case flags.FlagNote:
+			cmd.Println(tcc.Note)
+		default:
+			err := errUnknownConfigKey(key)
+			return fmt.Errorf("couldn't get the value for the key: %v, error:  %v", key, err)
+		}
+
+	case 2:
+		// it's set
+		key, value := args[0], args[1]
+
+		switch key {
+		case flags.FlagChainID:
+			tcc.ChainID = value
+		case flags.FlagKeyringBackend:
+			tcc.KeyringBackend = value
+		case tmcli.OutputFlag:
+			tcc.Output = value
+		case flags.FlagNode:
+			tcc.Node = value
+		case flags.FlagBroadcastMode:
+			tcc.BroadcastMode = value
+		case flags.FlagGas:
+			tcc.Gas = value
+		case flags.FlagGasPrices:
+			tcc.GasPrices = value
+			tcc.Fees = "" // resets since we can only use 1 at a time
+		case flags.FlagGasAdjustment:
+			tcc.GasAdjustment = value
+		case flags.FlagFees:
+			tcc.Fees = value
+			tcc.GasPrices = "" // resets since we can only use 1 at a time
+		case flags.FlagFeeGranter:
+			tcc.FeeGranter = value
+		case flags.FlagFeePayer:
+			tcc.FeePayer = value
+		case flags.FlagNote:
+			tcc.Note = value
+		default:
+			return errUnknownConfigKey(key)
+		}
+
+		confFile := filepath.Join(configPath, "client.toml")
+		if err := writeConfigToFile(confFile, &tcc); err != nil {
+			return fmt.Errorf("could not write client config to the file: %v", err)
+		}
+
+	default:
+		panic("cound not execute config command")
```

Comment:
can we update it to show how to use the command instead?

Reply:
this default is a placeholder, and really wont be accessed.